cache/src/main/java/io/envoyproxy/controlplane/cache/SnapshotCache.java

```diff
@@ -17,19 +17,19 @@
   boolean clearSnapshot(T group);
 
   /**
-   * Returns the most recently set {@link Snapshot} for the given node group.
+   * Returns the most recently set {@link V2Snapshot} for the given node group.
    *
    * @param group group identifier
    * @return latest snapshot
    */
-  Snapshot getSnapshot(T group);
+  U getSnapshot(T group);
 
   /**
-   * Set the {@link Snapshot} for the given node group. Snapshots should have distinct versions and be internally
+   * Set the {@link V2Snapshot} for the given node group. Snapshots should have distinct versions and be internally
    * consistent (i.e. all referenced resources must be included in the snapshot).
    *
    * @param group group identifier
    * @param snapshot a versioned collection of node config data
    */
-  void setSnapshot(T group, Snapshot snapshot);
+  void setSnapshot(T group, U snapshot);
```

Comment:
Callers will need to update to be specific about the type of snapshot they're generating, either V2Snapshot or V3Snapshot